### PR TITLE
feat: refactor WordXmlEscape and HtmlEscape

### PR DESCRIPTION
**Pull request by kienmatu**
## Summary
This PR refactors the `WordXmlEscape` and `HtmlEscape` classes to improve code readability, maintainability, and efficiency. Additionally, redundant escape logic has been removed since the `DocUtil.getEscapeAndCleanComment` method already handles the required escaping. Unit tests have been added and updated to ensure proper functionality and coverage.

### Refactored WordXmlEscape:

- Simplified the call method by removing redundant escape logic.
- Retained only the necessary replacements before invoking DocUtil.getEscapeAndCleanComment.

### Refactored HtmlEscape:

- Removed redundant escape logic since DocUtil.getEscapeAndCleanComment already performs the required escaping.
- Improved the readability of the call method.

### Unit Tests:

- Added new unit tests for edge cases in `WordXmlEscape` and `HtmlEscape`.
- Updated existing tests to reflect the removal of redundant escape logic.

### CODE REVIEW

**Comment by linwumingshi** [comment]
@kienmatu  Please use the `mvn spring-javaformat:apply` command to format the code, and then use `git commit --amend ` or another appropriate Git operation to combine the changes into a single commit.


# Contributing

see [contributing](https://smart-doc-group.github.io/guide/community/contributing)

**Comment by kienmatu** [comment]
Hi @linwumingshi, I have just already squashed the commits and reformat following spring rules. Thanks

File: src/main/java/com/ly/doc/function/HtmlEscape.java [hunk 1]
```diff
@@ -25,19 +25,49 @@
...
 /**
- * beetl template function
+ * Function to escape HTML content in Beetl templates.
+ * <p>
+ * This class provides a method to replace specific HTML characters with their
+ * corresponding escape sequences and clean up comments.
+ * <p>
+ * For example, it replaces:
+ * <p>
+ * - <code>&</code> with <code>&amp;amp;</code>
```
Comment (linwumingshi) [inline src/main/java/com/ly/doc/function/HtmlEscape.java]: - **Tip**: Use `{@code ...}` tags to avoid manual escaping:

  ```java
   {@code &} becomes {@code &amp;}
  ```

File: src/main/java/com/ly/doc/function/HtmlEscape.java [hunk 2]
```diff
@@ -25,19 +25,49 @@
...
+ * <p>
+ * - <code>"</code> with <code>&amp;quot;</code>
```
Comment (linwumingshi) [inline src/main/java/com/ly/doc/function/HtmlEscape.java:39]: **Tip**: Use `{@code ...}` tags to avoid manual escaping:

**Comment by kienmatu** [comment]
ci run failed, so I updated the javadoc then It passed locally. Please help to check again @linwumingshi, thanks.

**Review by linwumingshi — APPROVED** [review]
